libraries/vircadia-client/src/client.cpp

```diff
@@ -0,0 +1,154 @@
+//
+//  client.cpp
+//  libraries/client/src
+//
+//  Created by Nshan G. on 15 March 2022.
+//  Copyright 2022 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "client.h"
+
+#include <list>
+
+#include <nlohmann/json.hpp>
+
+#include "version.h"
+#include "internal/Context.h"
+
+namespace vircadia { namespace client
+{
+    std::list<Context> contexts;
+}} // namespace vircadia::client
+
+
+using namespace vircadia::client;
+
+VIRCADIA_CLIENT_DYN_API
+vircadia_context_params vircadia_context_defaults() {
+    return {
+        -1, -1,
+        nullptr,
+        nullptr,
+        {
+            "VircadiaClient",
+            "Vircadia",
+            "vircadia.com",
+            vircadia_client_version()->full
+        }
+    };
+}
+
+VIRCADIA_CLIENT_DYN_API
+int vircadia_create_context(vircadia_context_params params) {
```

Comment:
Is the use of "_"s in names a C or Unity convention?
It is very different to the normal Vircadia naming convention.

Reply:
It's a common C language convention, and I went it to make the API function stand out. They shouldn't really be used anywhere in the main repo, other than the unit tests (which also kind of feel like C examples). They also stand out in Unity where the most things are camel case also.

Reply:
Perhaps some comments could be added to the README to provide guidance to other devs:
- File naming (C API vs. C++ internals)
- Variable naming (C API vs. C++ internals)
- The use of the /internal subdirectory.
- Naming in unit tests.

Reply:
Added an explanation in the readme here https://github.com/vircadia/vircadia/blob/18e13235d6a893593492a447c8ab5beb7f6fa757/libraries/vircadia-client/README.md

Reply:
Looks good, thanks.